### PR TITLE
Implement edit,complete status and clear all functionality

### PR DESCRIPTION
**Pull request by Elhambasir**
1. Use the jest framework for testing.
2. Create a test file(s) ([..].test.js) for a file(s) containing the following functions:
    - a function for editing the task description.
    - a function for updating an item's 'completed' status.
    - the "Clear all completed" function.
3.  Group The tests using the `describe()` method.

### CODE REVIEW

**Review by BenMukebo — CHANGES_REQUESTED** [review]
..

File: README.md [hunk 1]
```diff
@@ -215,7 +215,8 @@ This project is deployed at github pages you can clone it here. [Please click to
 - Linkedin: [@linkedinhandle](https://www.linkedin.com/in/basir-ahmadi-8aba89263)
 - Email: elham1378basir@gmail.com
   <br>
```
Comment (BenMukebo) [inline README.md:217]: - [ ] Your [Live Demo Link](https://elhambasir.github.io/To-Do-list/) is boken, Kindly fix this or remove it 👍🏻

**Review by BenMukebo — CHANGES_REQUESTED** [review]
### Hi @Elhambasir, @Lul-Abdifan,

You did a great job so far 👏🏼 👏🏼
There are some issues that you still need to work on to go to the next project but you are almost there!
<FEEL FREE TO CHANGE THIS MESSAGE TO YOUR PERSONAL WORDS>

### To highlight 💪🏻
- GitHub flow was correctly used.  ✔️   
- Descripitive README ✔️ 
- Use mock for functions that used `localStorage`. ✔️ 
- All functions for this mistone are tested.  ✔️ 
 👍  Good job!


## STATUS: CHANGES REQUESTED  ♻️ 
_Please Check the comments under the review and make all required changes_

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏 💻  
Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please ping me @BenMukebo when you comment so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: README.md (hunk 1 above)
Comment (Elhambasir) [reply README.md:217]: I fixed that. Thank you.

**Review by binhussen — APPROVED** [review]
Hi @Elhambasir , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉

## Highlights
- All linter checks passed ✔️
- Descriptive pull request ✔️
- Used GitHub flow ✔️
## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**